File: go-client-sdk/pkg/models/shared/codatpublicapimodelscompanyprofilemodel.go
package shared

type CodatPublicAPIModelsCompanyProfileModel struct {
	AlertAuthHeader    *string  `json:"alertAuthHeader,omitempty"`
	APIKey             *string  `json:"apiKey,omitempty"`
	ConfirmCompanyName *bool    `json:"confirmCompanyName,omitempty"`
	IconURL            *string  `json:"iconUrl,omitempty"`
	LogoURL            *string  `json:"logoUrl,omitempty"`
	Name               *string  `json:"name,omitempty"`
	RedirectURL        *string  `json:"redirectUrl,omitempty"`
	WhiteListUrls      []string `json:"whiteListUrls,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelscompanyprofilemodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose } from "class-transformer";


export class CodatPublicApiModelsCompanyProfileModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "alertAuthHeader" })
  alertAuthHeader?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "apiKey" })
  apiKey?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "confirmCompanyName" })
  confirmCompanyName?: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "iconUrl" })
  iconUrl?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "logoUrl" })
  logoUrl?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "name" })
  name?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "redirectUrl" })
  redirectUrl?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "whiteListUrls" })
  whiteListUrls?: string[];
}